src/navigations/root-stack.js

```diff
@@ -131,11 +133,21 @@ const AuthenticatedStack = createNativeStackNavigator();
 
 const withSafeAreaView = (Component) => {
   return (props) => (
-    <SafeAreaView style={{flex: 1, backgroundColor: 'white'}}>
+    <SafeAreaView style={{flex: 1, backgroundColor: COLORS.almostBlack}}>
       <Component {...props} />
     </SafeAreaView>
   );
 };
+const withSafeAreaViewTopBottom = (Component, topColor, bottomColor = COLORS.almostBlack) => {
+  return (props) => (
+    <React.Fragment>
+      <SafeAreaView style={{flex: 0, backgroundColor: topColor}} />
+      <SafeAreaView style={{flex: 1, backgroundColor: bottomColor}}>
+        <Component {...props} />
+      </SafeAreaView>
+    </React.Fragment>
+  );
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `withSafeAreaView` function has been updated to use `COLORS.almostBlack` instead of `'white'`. A new function `withSafeAreaViewTopBottom` has also been added which allows different colors for the top and bottom SafeAreaView. These changes are consistent with the overall theme update.

```diff
- <SafeAreaView style={{flex: 1, backgroundColor: 'white'}}>
+ <SafeAreaView style={{flex: 1, backgroundColor: COLORS.almostBlack}}>
+ const withSafeAreaViewTopBottom = (Component, topColor, bottomColor = COLORS.almostBlack) => {
+   return (props) => (
+     <React.Fragment>
+       <SafeAreaView style={{flex: 0, backgroundColor: topColor}} />
+       <SafeAreaView style={{flex: 1, backgroundColor: bottomColor}}>
+         <Component {...props} />
+       </SafeAreaView>
+     </React.Fragment>
+   );
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->